src/Raven.Server/ServerWide/ClusterCommandsVersionManager.cs

```diff
@@ -174,11 +177,11 @@ public static void SetClusterVersion(int version)
                     break;
                 Interlocked.CompareExchange(ref _currentClusterMinimalVersion, version, currentVersion);
             }
-
-            if (currentVersion != version && Log.IsInfoEnabled)
-            {
+            
+            if (Log.IsInfoEnabled && previousVersion != version) 
                 Log.Info($"Cluster version was changed from {currentVersion} to {version}");
-            }
+
+            AfterClusterVersionSetting?.Invoke(null, new AfterClusterVersionSettingEventArgs(previousVersion, currentVersion));
```

Comment:
invoke it only if version actually changed, so inside the `if` I guess?

Reply:
Apparently, I'm misunderstanding something somewhere, so I want to outline all the movements of my thoughts to identify the place where my approach is incorrect.

The crux of the initial issue is that during the `ServerStore` initialization, we encounter a race condition. Specifically, we are trying to execute `PutMyNodeInfoAsync` at a time when the `RachisConsensus` isn't fully formed and the cluster lacks an assigned `version` value.

To navigate this problem, we decided to capture the initial assignment of the version (transitioning from `zero` to the `current` version) within the version setting method. After this, we execute `PutMyNodeInfoAsync` just once only on the version transitioning from zero, and does not return to it, as it is unnecessary to update `NodeInfo` every time the cluster version changes.

Then, during the initialization process in `LicenseManager`, we subscribe to this event only if we don't yet have a cluster version at this stage of `LicenseManager` initialization. We then unsubscribe from this event after `PutMyNodeInfoAsync` has been executed once. However, if the cluster version is already set to the moment of `LicenseManager` initialization, we bypass the event subscription and immediately execute `PutMyNodeInfoAsync` since nothing prevents us from doing this.

And now to the answer to the above question: 
Yes, if the cluster version changes, we execute `OnClusterVersionChange?.Invoke()` and log the corresponding message if necessary.